mtags/src/main/scala-2/scala/meta/internal/pc/SignatureHelpProvider.scala

```diff
@@ -489,12 +489,16 @@ class SignatureHelpProvider(val compiler: MetalsGlobal) {
       case TypeRef(
             _,
             cls,
-            TypeRef(_, _, args) :: Nil
+            TypeRef(_, symbol, args) :: Nil
           )
           if isUnapplyMethod &&
-            (cls == definitions.OptionClass || cls == definitions.SomeClass) &&
-            args.nonEmpty =>
-        List(args.map(_.typeSymbol))
+            (cls == definitions.OptionClass || cls == definitions.SomeClass) =>
+        // tuple unapply results
+        if (args.nonEmpty && definitions.isTupleSymbol(symbol))
```

Comment:
[nitpick] Do we need `args.nonEmpty` ? I believe there's no tuple with no arguments.

```suggestion
        if (definitions.isTupleSymbol(symbol))
```

Reply:
probably not, I wanted to be too safe :sweat_smile: